extension/sql/timescale-prometheus.sql

```diff
@@ -159,4 +159,78 @@ AS $func$
     VALUES ('timescale_prometheus_' || meta_key,meta_value, send_telemetry)
     ON CONFLICT (key) DO UPDATE SET value = EXCLUDED
 $func$
-LANGUAGE SQL VOLATILE SECURITY DEFINER;
\ No newline at end of file
+LANGUAGE SQL VOLATILE SECURITY DEFINER;
+
+----------------------------  aggregation functions ----------------------------
+
+CREATE FUNCTION @extschema@.prom_delta_transition(state internal, lowest_time timestamptz,
+    greatest_time timestamptz, step bigint, range bigint,
+    timev timestamptz, val double precision
+    )
+RETURNS internal AS '$libdir/timescale_prometheus_extra', 'gapfill_delta_transition'
+LANGUAGE C IMMUTABLE PARALLEL SAFE;
+
+CREATE FUNCTION @extschema@.prom_rate_transition(state internal, lowest_time timestamptz,
+    greatest_time timestamptz, step bigint, range bigint,
+    timev timestamptz, val double precision
+    )
+RETURNS internal AS '$libdir/timescale_prometheus_extra', 'gapfill_rate_transition'
+LANGUAGE C IMMUTABLE PARALLEL SAFE;
+
+CREATE FUNCTION @extschema@.prom_increase_transition(state internal, lowest_time timestamptz,
+    greatest_time timestamptz, step bigint, range bigint,
+    timev timestamptz, val double precision
+    )
+RETURNS internal AS '$libdir/timescale_prometheus_extra', 'gapfill_increase_transition'
+LANGUAGE C IMMUTABLE PARALLEL SAFE;
+
+CREATE FUNCTION @extschema@.prom_extrapolate_final(state internal)
+RETURNS DOUBLE PRECISION[]
+AS '$libdir/timescale_prometheus_extra', 'gapfill_delta_final'
+LANGUAGE C IMMUTABLE PARALLEL SAFE;
+
+
+-- implementation of prometheus delta function
+-- for proper behavior the input must be ORDER BY timev
+CREATE AGGREGATE @extschema@.prom_delta(
+    lowest_time timestamptz,
+    greatest_time TIMESTAMPTZ,
+    step BIGINT,
+    range BIGINT,
+    timev TIMESTAMPTZ,
```

Comment:
nit maybe data_time and data_value as arg parameters. `timev` is just weird and this is public-facing api

Reply:
bleh, don't love that either, do you prefer `data_time` or `"time"`?